src/fn/upload-app-settings.js

```diff
@@ -1,14 +1,21 @@
 const api = require('../lib/api');
-const environment = require('../lib/environment');
 const fs = require('../lib/sync-fs');
+const path = require('path');
+const { info, error } = require('../lib/log');
+const environment = require('../lib/environment');
 
-module.exports = () => {
-  const settings = fs.read(`${environment.pathToProject}/app_settings.json`);
+module.exports = async () => {
+  const settings = fs.read(path.join(environment.pathToProject, 'app_settings.json'));
   return api().updateAppSettings(settings)
-    .then(JSON.parse)
-    .then(json => {
-      // As per https://github.com/medic/medic-webapp/issues/3674, this endpoint
-      // will return 200 even when upload fails.
-      if(!json.success) throw new Error(json.error);
+    .then(res => {
+      info(`app_settings uploaded successfully: ${res}`);
```

Comment:
Can I ask about the change in behavior here? Does it still work with legacy versions of the webapp? Shouldn't we still JSON.parse the result and confirm it has a truthy success attribute for those versions?

Reply:
2.18.1 is supported in that matrix. So this feedback is still relevant.

Reply:
I've realized that since no functionality will have to be traded to add legacy support, it's easier to add it back and not have to worry about unsupported versions.